app/assets/javascripts/govuk_publishing_components/components/search-autocomplete.js

```diff
@@ -0,0 +1,318 @@
+/* global fetch */
+
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {}
+window.GOVUK.Modules.GovukAutocomplete = window.GOVUKFrontend.Autocomplete;
+
+(function (Modules) {
+  'use strict'
+
+  class GemSearchAutocomplete {
+    constructor ($module) {
+      this.$module = $module
+
+      this.numberSuggestions = $module.getAttribute('data-display-number-suggestions') || 5
+      this.throttleDelayTime = $module.getAttribute('data-throttle-delay-time') || 0
+      this.submitOnSelect = $module.getAttribute('data-submit-on-select') || false
+      this.baseClass = $module.getAttribute('data-base-class') || 'gem-c-search-autocomplete'
+      this.source = $module.getAttribute('data-source')
+    }
+
+    init () {
+      this.onUpdate = this.handleUpdate
+      this.onSubmit = () => { }
+
+      this.expanded = false
+      this.loading = false
+
+      this.$module.insertAdjacentHTML('beforeend', '<ul class="gem-c-search-autocomplete__result-list govuk-body" role="listbox"></ul><div aria-atomic="true" aria-live="polite" role="status" class="govuk-visually-hidden">No results.</div>')
+
+      this.$resultList = this.$module.querySelector('ul')
+      this.$resultList.setAttribute('role', 'listbox')
```

Comment:
I'm not sure why we'd add this here rather than have it in the block of HTML, that would make things shorter. There's no reason this needs to be dynamically added.